src/net.cpp

```diff
@@ -2830,18 +2830,17 @@ void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
         pnode->mapSendBytesPerMsgCmd[msg.m_type] += nTotalSize;
         pnode->nSendSize += nTotalSize;
 
-        if (pnode->nSendSize > nSendBufferMaxSize)
-            pnode->fPauseSend = true;
+        if (pnode->nSendSize > nSendBufferMaxSize) pnode->fPauseSend = true;
+        if (gArgs.IsArgSet("-logmessages")) {
+            LogMessage(*pnode, msg.m_type, MakeSpan(msg.data), /* incoming */ false);
```

Comment:
Could this be moved to the start of the function right after the `LogPrint(BCLog::NET, …)` call to a.) avoid being run while we're holding `pnode->cs_vSend` (which we don't need for logging), and b.) make sure all `sending %s (%d bytes) peer=%d``BCLog::NET` debug log entries are guaranteed to have a corresponding entry in the message capture file (in the absence of failed logged))?

Reply:
Yup, it can, and I agree with you that it should be.  The locking in PushMessage is a bit of a mess, which I fix up a little in #19673.